test/python/circuit/test_gate_labels.py

```diff
@@ -0,0 +1,17 @@
+from test import QiskitTestCase
+import numpy as np
+from qiskit import QuantumCircuit
+
+
+class TestGateLabels(QiskitTestCase):
+    def test_gate_labels_are_applied(self):
```

Comment:
Could we put this test function in an existing class in `test/python/circuit/test_circuit_operations.py`, just to avoid an extra stub file?  There's no _perfect_ place for it, but I think we can avoid the new file.